### ISSUE TITLE
[SOLVED]Stuck in loading emulator screen / Nesbox doesn't work on local html!

### ISSUE THREAD

**Issue by Strike-9344** [nesbox/emulator#100]
### Problem: Stuck in the "loading emulator 100% screen"

I'm one week away from finishing my programming course in my school, and my final project was a gaming site that the visitors could run old games on-line. So, right now, **I only have one week to make this work**, however, nesbox isn't working and **the situation is pretty desperate** because I don't understand why things aren't working at all.

First, I'm not sure if in the presentation day, students will have internet, so the site is local. Right now, I'm trying to place nesbox in the site, **which is the most important part of the project.**

However, even though I'm only changing in the code nesbox.swf location path and the flashvars variable, it still doesn't work.

Before I'm implement nesbox on the project's site, I'm testing it on a isolated html first to make sure it works. Here it is:

`<!doctype html>
<head><title>NESBOX</title></head>
<body>

<div>
	<div id="emulator">
		<p>To play this game, please, download the latest Flash player!</p>
		<br>
		<a href="http://www.adobe.com/go/getflashplayer">
			<img src="//www.adobe.com/images/shared/download_buttons/get_adobe_flash_player.png" alt="Get Adobe Flash player"/>
		</a>
	</div>
</div>

<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.9.0/jquery.min.js"></script>
<script src="http://ajax.googleapis.com/ajax/libs/swfobject/2.2/swfobject.js"></script>

<script type="text/javascript">

	var resizeOwnEmulator = function(width, height)
	{
		var emulator = $('#emulator');
		emulator.css('width', width);
		emulator.css('height', height);
	}

	$(function()
	{
		function embed()
		{
			var emulator = $('#emulator');
			if(emulator)
			{
				var flashvars = 
				{
					system : 'nes',
					url : 'roms/Mega Man (USA).nes'
				};
				var params = {};
				var attributes = {};
				
				params.allowscriptaccess = 'sameDomain';
				params.allowFullScreen = 'true';
				params.allowFullScreenInteractive = 'true';
				
				swfobject.embedSWF('bin/Nesbox.swf', 'emulator', '640', '480', '11.2.0', 'flash/expressInstall.swf', flashvars, params, attributes);
			}
		}
		
		embed();
	});
	
</script>

</body>
</html>`

When I downloaded emulator-master.zip, I just extracted the zip and put the html file inside, so Nesbox.swf is in the bin folder and the game is in the roms folder(That I created, btw).

Even though all the paths are correct, it's still not working. The only reason I could think is that nesbox doesn't work with local file-paths, but I'm not sure because the emulator does run, however, it still stuck in the "loading emulator 100%" screen.

![image](https://user-images.githubusercontent.com/58578020/70289186-cf6f0b00-17b2-11ea-963c-8767b32f85f1.png)

Here's the code in txt in case you may need. All you need to do is save in html to run.
[nesbox test.txt](https://github.com/nesbox/emulator/files/3930015/nesbox.test.txt)

**Comment by nesbox** [comment]
Do you see any errors in the browser console?

**Comment by Strike-9344** [comment]
> Do you see any errors in the browser console?

Only a "file not found" about a Adobe Flash Player image that appears when someone doesn't allow Flash or doesn't have Flash installed in their computer. But I'm using chrome, so flash comes with the browser so it's probably not related to the problem.

![image](https://user-images.githubusercontent.com/58578020/70328179-345c4c80-1817-11ea-8a2c-a7652bb3fec4.png)

**Comment by Strike-9344** [comment]
> > Do you see any errors in the browser console?
> 
> Only a "file not found" about a Adobe Flash Player image that appears when someone doesn't allow Flash or doesn't have Flash installed in their computer. But I'm using chrome, so flash comes with the browser so it's probably not related to the problem.
> 
> ![image](https://user-images.githubusercontent.com/58578020/70328179-345c4c80-1817-11ea-8a2c-a7652bb3fec4.png)

Here's the address of the image that's been not found:
file://www.adobe.com/images/shared/download_buttons/get_adobe_flash_player.png

![image](https://user-images.githubusercontent.com/58578020/70328365-9ae16a80-1817-11ea-81cd-78905563ed17.png)

**Comment by Strike-9344** [comment]
And now the only error in the browser console is gone because almost every src in the original code has "//" in the start of the source-path, so chrome doesn't recognize it as a web-address(It recognizes as a "file://" address), so I have to put "http:" for the image to work(Same thing happened with the script src links - Which is why, at the first run, the emulator wasn't even visible).

![image](https://user-images.githubusercontent.com/58578020/70328554-17744900-1818-11ea-8f3b-5a5bf0019b62.png)

**Comment by Strike-9344** [comment]
I just found out what the problem exactly was.
**Nesbox doesn't run in local html pages.**

I installed XAMPP and ran the _nesboxteste.html_ with the localhost address and it worked.
![image](https://user-images.githubusercontent.com/58578020/70367304-71f3c080-187d-11ea-8e6f-87928cf317a4.png)

I tried other javascript emulators and had the same problem and now all of them are working. So, **every javascript/flash emulator doesn't work in html local pages.**

I suggest @nesbox to put an warning in the readme file about that(Like: warning: only work in servers/doesn't work in local files), because I really wasn't expecting this kind of problem.
